### ISSUE TITLE
Support eslint.config.cjs

### ISSUE THREAD

**Issue by valtlai** [eslint/eslint#14170]
**The problem you want to solve.**

The upcoming simplified configuration system ([RFC](/eslint/rfcs/tree/master/designs/2019-config-simplification)), or flat config (#13481), only supports `eslint.config.js`, but it can’t be used if there’s `"type": "module"` in the project’s `package.json`, since ESLint doesn’t currently support loading an ECMAScript module as a config file.

**Your take on the correct solution to problem.**

ESLint should also support `eslint.config.cjs` by default, so that `"type": "module"` can be used along with ESLint.

**Are you willing to submit a pull request to implement this change?**

Maybe if it’s not too hard to implement.

**Comment by mdjermanovic** [comment]
Hi @valtlai, thanks for the issue!

Would it be necessary if we manage to support ESM configs?

**Comment by valtlai** [comment]
@mdjermanovic That would be even better and then the support for `.cjs` isn’t needed, I think.

**Comment by aladdin-add** [comment]
👍 it's no harm to support `eslint.config.js`, and also `eslint.config.mjs`

**Comment by valtlai** [comment]
So either `eslint.config.mjs` or `eslint.config.cjs` should be supported, depending on whether ESM configs are supported or not.

In the former case, `eslint.config.cjs` may be supported too, but I believe it wouldn’t be necessary, because you can import CJS modules and therefore use ESLint plugins written in CJS in your ESM config.

**Comment by mdjermanovic** [comment]
We're thinking about using `import()` to load `eslint.config.js`. In that case, I think that support for `.cjs` and `.mjs` extensions doesn't seem necessary: in `"type": "module"` projects `eslint.config.js` should be written as ESM, in other projects as CJS.

If it turns out that we still have to use `require()` to load configs, then I think we should support `eslint.config.cjs`, because `require("eslint.config.js")` doesn't work in `"type": "module"` projects.

**Comment by valtlai** [comment]
@mdjermanovic Yeah, you’re very correct. I was  mostly wrong in my previous comment.

However, it’s possible to import CJS modules, so `require()` and therefore `.cjs` shouldn’t be needed.

**Comment by nzakas** [comment]
Thanks for opening this issue. I think it’s too early in the development of the new config system to say if this would be useful or necessary, so I’d like to table the discussion until we at least have a working prototype.

let’s leave this issue open so we can circle back at that time.

**Comment by DamienCassou** [comment]
Is there any workaround to have eslint working in `type:"module"` projects?

**Comment by aladdin-add** [comment]
you can use `.eslintrc.cjs`.

**Comment by Shinigami92** [comment]
What about just using https://github.com/davidtheclark/cosmiconfig ?

**Comment by bmish** [comment]
For the record, this has been addressed in:
* https://github.com/eslint/eslint/pull/17909
